### ISSUE TITLE
`execute-command` is slower by a significant factor

### ISSUE THREAD

**Issue by aadcg** [atlas-engineer/nyxt#2720]
I'm measuring the time in a rather clumsy way but it gets the job done for now.

- `(let ((*interactive-p* t)) (time (execute-command)))`
- hit `RET` as soon as the prompt buffer appears on the screen

Here are the results:

**3-pre-release-2**
```log
Evaluation took:
  0.708 seconds of real time
  0.286832 seconds of total run time (0.265672 user, 0.021160 system)
  40.54% CPU
  10 lambdas converted
  1,565,051,424 processor cycles
  8,370,912 bytes consed  
```

**master**
```log
Evaluation took:
  1.292 seconds of real time
  0.982262 seconds of total run time (0.959376 user, 0.022886 system)
  76.01% CPU
  20 lambdas converted
  2,850,394,976 processor cycles
  74,934,592 bytes consed
```

Investigating.

**Comment by Ambrevar** [comment]
To reproduce this you must make ensure the current buffer is visiting the same page.

In particular, visiting a duckduckgo page with JS enabled slows down the prompt buffer dramatically.  This is a known issue and it's probably a WebKitGTK bug.

Also to benchmark non-interactively you can wait for the prompt buffer to be ready, there is `(nyxt:prompt-buffer-ready-hook nyxt:*browser*)`.

**Comment by aadcg** [comment]
Inconclusive. Closing.

**Comment by Ambrevar** [comment]
? What happened?

**Comment by aadcg** [comment]
I couldn't get consistent results that indicate that between `master` and `3-pre-release-2`, `execute-command` takes more time to pop up.

**Comment by Ambrevar** [comment]
OK :)

**Comment by aadcg** [comment]
Fixed via #2722.
